Upgrade to Django 3.0

juliorkm opened the pull request:
**django-all-access** unfortunately only supports Django 1. Many projects have difficulty upgrading to newer Django versions due to this dependency. This pull request makes this project work on Django 3.0, but due to Django's lack of backward compatibility, removes the ability to use it on earlier versions.

Commit:
Fixed from_db_value method definition

Commit:
Removed python2 support in models.py

Commit:
Fixed import of Django's reverse method

Commit:
Added backend to login() calls

Commit:
Added 'request' parameter to authenticate() method

Commit:
Updated Django version to 3.0.3 and fixed runtests.py

Commit:
Updated setup.py

sergios committed:
Included Makefile

mlavin commented:
I appreciate this effort and it would be great to see this project working on the latest Django (though as you can probably tell I don't really use this very much anymore). But I can't justify merging this while the test suite is broken.

Commit:
Fixed test suite

juliorkm commented:
My apologies, I had already fixed the tests for the most recent Django version, but hadn't updated the tox and travis configuration files. The test suite is now fixed.

mlavin commented on setup.py:
```diff
@@ -13,31 +13,29 @@ def read_file(filename):
 
 setup(
-    name='django-all-access',
+    name='django3-all-access',
```
You are obviously welcome to publish your own version of this package under the terms of the BSD license but that shouldn't be part of this PR.

bjinwright commented:
For those coming to this later. The new project from @juliorkm is
https://github.com/stormers/django-all-access. Mark @mlavin thanks for releasing the project and I hope you're doing well buddy.

mlavin commented:
Brian! I'm doing well (all things considered). Sadly I don't really use Django much anymore, in case that wasn't already obvious from my inactivity here. If @juliorkm or you want to take over this project I'm happy to get you access to the PyPi repository and Read the Docs ❤️
